CommonTools/MVAUtils/src/GBRForestTools.cc

```diff
@@ -117,6 +120,16 @@ namespace {
   }
 
   std::unique_ptr<GBRForest> init(const std::string& weightsFileFullPath, std::vector<std::string>& varNames) {
+    //
+    // Load weights file, for ROOT file
+    //
+    if (reco::details::hasEnding(weightsFileFullPath, ".root")) {
+      TFile gbrForestFile(weightsFileFullPath.c_str());
+      // Use the copy-constructor of GBRForest to copy the GBRForest.
+      // In this way, the ROOT file can be closed.
+      return std::make_unique<GBRForest>(*(GBRForest*)gbrForestFile.Get("gbrForest"));
```

Comment:
```suggestion
      // the pointer is free to be taken over
      return std::unique_ptr<GBRForest>(reinterpret_cast<GBRForest*>(gbrForestFile.Get("gbrForest")));
```

based on what I see in https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_11_2_0_pre9-sign1110.136.88811.step2.IgProf.1.MEM_LIVE/82
the forest read from the file is still in memory, in addition to the copy.
There is an example in RecoEgamma/EgammaElectronAlgos/src/RegressionHelper.cc where the forests are not copied after reading from file.

https://mattermost.web.cern.ch/cms-o-and-c/pl/z5e5yqqqetd75ded9zi3bqfcay
may have a more certain confirmation (not yet as of this post submission).

after the update, the total is a bit more acceptable
https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_11_2_0_pre9-sign1110-1fc9343.136.88811.step2.IgProf.1.MEM_LIVE/175

#32293 is made to keep track of possible improvement in size